### ISSUE TITLE
incorrect error reported for numpy

### ISSUE THREAD

**Issue by lhfriedman** [PyCQA/pylint#3640]
### Steps to reproduce
1. Use this test code in file flat_iter.py

```
    import numpy as np
    a_0 = np.array([[1, 2, 3]])
    print(a_0)
    a_0.flat[1] = 2**14
    print(a_0)
```
>    [[1 2 3]]
>    [[    1 16384     3]]

2. 

> pylint flat_iter.py
************* Module flat_iter
flat_iter.py:14:0: E1137: 'a_0.flat' does not support item assignment (unsupported-assignment-operation)

-----------------------------------
Your code has been rated at 0.00/10

### Current behavior
reports E1137

### Expected behavior
ignores correct python code

### pylint --version output

> pylint 2.5.0
> astroid 2.4.0
> Python 3.7.7 (default, May  6 2020, 11:45:54) [MSC v.1916 64 bit (AMD64)]

**Comment by brycepg** [comment]
For some reason numpy is inferring the `array` as an `ndarray`
```
import astroid

a_0_astroid = astroid.extract_node(
"""
import numpy as np
a_0 = np.array([[1, 2, 3]])
print(a_0)
a_0.flat[1] = 2**14
a_0 #@
""")
[nd_astroid] = a_0_astroid.inferred()
print(nd_astroid)
```
`<Instance of .ndarray at 0x2253675315944>`

If someone wanted to check how `numpy.array` is assigned in the `numpy` source code, that may help us understand how `astroid` is messing up.

**Comment by hippo91** [comment]
@brycepg ,

the problem is due to the fact that the `flat` attribute of `numpy.ndarray` is said to be `None` in the corresponding brain. (`brain_numpy_ndarray.py`).
I'll try to fix this ASAP.
